cmd/pitr/pxc/pxc.go

```diff
@@ -330,6 +302,45 @@ func getBinlogTimeByName(ctx context.Context, db *PXC, binlogName string) (int64
 	return binlogTime, nil
 }
 
+func (p *PXC) InstallBinlogUDFComponent(ctx context.Context) error {
+	var urn string
+	component := p.db.QueryRowContext(ctx, "SELECT component_urn FROM mysql.component WHERE component_urn = 'file://component_binlog_utils_udf'")
+	if err := component.Scan(&urn); err != nil && !errors.Is(err, sql.ErrNoRows) {
+		return errors.Wrap(err, "get component_binlog_utils_udf")
+	}
+
+	if len(urn) > 0 {
+		log.Printf("file://component_binlog_utils_udf is already installed")
+		return nil
+	}
+
+	_, err := p.db.ExecContext(ctx, "INSTALL COMPONENT 'file://component_binlog_utils_udf'")
+	if err != nil {
+		return errors.Wrap(err, "install component")
+	}
+
+	return nil
+}
+
+func (p *PXC) CreateCollectorFunctions(ctx context.Context) error {
+	_, err := p.db.ExecContext(ctx, "CREATE FUNCTION IF NOT EXISTS get_last_record_timestamp_by_binlog RETURNS INTEGER SONAME 'binlog_utils_udf.so'")
+	if err != nil {
+		return errors.Wrap(err, "create function get_first_record_timestamp_by_binlog")
+	}
+
+	_, err = p.db.ExecContext(ctx, "CREATE FUNCTION IF NOT EXISTS get_gtid_set_by_binlog RETURNS STRING SONAME 'binlog_utils_udf.so'")
+	if err != nil {
+		return errors.Wrap(err, "create function get_gtid_set_by_binlog")
+	}
+
+	_, err = p.db.ExecContext(ctx, "CREATE FUNCTION IF NOT EXISTS get_first_record_timestamp_by_binlog RETURNS INTEGER SONAME 'binlog_utils_udf.so'")
+	if err != nil {
+		return errors.Wrap(err, "create function get_first_record_timestamp_by_binlog")
+	}
+
+	return nil
+}
```

Comment:
If we want to avoid having partial creation of the functions described here, we can utilize `p.db.BeginTx` and either commit or of them or none.

Reply:
i don't think we need it. partial creation is fine. if creating any of these fail, collector will crash and create them in the next run.

Reply:
Yes I thought that it made much more sense not to crash partially here since these can be done atomically, but it is fine